src/pages/test-report/components/competency/__tests__/competency.spec.ts

```diff
@@ -89,139 +92,320 @@ describe('CompetencyComponent', () => {
           competency: component.competency,
           newFaultCount: 1,
         }));
       });
-
-      it('should not dispatch a ADD_DRIVING_FAULT action if dangerous mode is active', () => {
+      it('should not dispatch an ADD_DRIVING_FAULT action if dangerous mode is active', () => {
         component.competency = Competencies.clearance;
         component.isDangerousMode = true;
 
         const storeDispatchSpy = spyOn(store$, 'dispatch');
-        component.recordFault();
+        component.addOrRemoveFault(false);
 
         expect(storeDispatchSpy).not.toHaveBeenCalledWith(new AddDrivingFault({
           competency: component.competency,
           newFaultCount: 1,
         }));
       });
+    });
+
+    describe('addDangerousFault', () => {
+      it('should dispatch a ADD_DANGEROUS_FAULT action if dangerous mode is active on press', () => {
+        component.competency = Competencies.clearance;
+        component.isDangerousMode = true;
+
+        const storeDispatchSpy = spyOn(store$, 'dispatch');
+        component.addOrRemoveFault(true);
+
+        expect(storeDispatchSpy).toHaveBeenCalledWith(new AddDangerousFault(component.competency));
+        expect(storeDispatchSpy).toHaveBeenCalledWith(new ToggleDangerousFaultMode());
+      });
+      it('should dispatch a ADD_DANGEROUS_FAULT action if dangerous mode is active on press and hold', () => {
+        component.competency = Competencies.clearance;
+        component.isDangerousMode = true;
+
+        const storeDispatchSpy = spyOn(store$, 'dispatch');
+        component.addOrRemoveFault(false);
+
+        expect(storeDispatchSpy).toHaveBeenCalledWith(new AddDangerousFault(component.competency));
+        expect(storeDispatchSpy).toHaveBeenCalledWith(new ToggleDangerousFaultMode());
+      });
+      it('should not dispatch a ADD_DANGEROUS_FAULT action if there is a dangerous fault', () => {
+        component.competency = Competencies.clearance;
+        component.hasDangerousFault = true;
+        component.isDangerousMode = true;
+
+        const storeDispatchSpy = spyOn(store$, 'dispatch');
+        component.addOrRemoveFault(true);
+
+        expect(storeDispatchSpy).not.toHaveBeenCalledWith(new AddDangerousFault(component.competency));
+        expect(storeDispatchSpy).not.toHaveBeenCalledWith(new ToggleDangerousFaultMode());
+      });
+      it('should not dispatch a ADD_DANGEROUS_FAULT action if there is a dangerous fault', () => {
+        component.competency = Competencies.clearance;
+        component.hasDangerousFault = true;
+        component.isDangerousMode = true;
+
+        const storeDispatchSpy = spyOn(store$, 'dispatch');
+        component.addOrRemoveFault(true);
+
+        expect(storeDispatchSpy).not.toHaveBeenCalledWith(new AddDangerousFault(component.competency));
+        expect(storeDispatchSpy).not.toHaveBeenCalledWith(new ToggleDangerousFaultMode());
+      });
+    });
```

Comment:
duplicated test for ADD_DANGEROUS_FAULT action if there is a dangerous fault

Reply:
yup looks that way.. nice spot